### ISSUE TITLE
fix: support deserialization of all known examples

### ISSUE THREAD

**Issue by jkowalleck** [CycloneDX/cyclonedx-python-lib#537]
in https://github.com/CycloneDX/specification/tree/master/tools/src/test/resources 
there are some example data, that do not work wit hthe deserialization of this library.

lets fix this

**Comment by madpah** [comment]
@jkowalleck - if I'm reading the current code in `main` correctly, this is already covered by the work you did to add `schemaTestData` used in `test_validation_json.py` and `test_validation_xml.py`?

**Comment by jkowalleck** [comment]
> [...] this is already covered by the work you did to add `schemaTestData` used in `test_validation_json.py` and `test_validation_xml.py`?

Nope. this issue is not about validating test data, but about deserializing the test data.
see how https://github.com/CycloneDX/cyclonedx-python-lib/pull/538 failed?

**Comment by madpah** [comment]
Gotcha! Will work on #581 and #422

**Comment by jkowalleck** [comment]
related: #600

**Comment by jkowalleck** [comment]
:mega: **this issue is blocked by**

- #422 
- #561 
- #581
